streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -25,6 +25,7 @@
 import org.apache.kafka.clients.admin.RecordsToDelete;
 import org.apache.kafka.clients.consumer.CommitFailedException;
 import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
```

Comment:
Could we add a few more unit tests for:

1. When handling Revocation, we commit all tasks.
2. When handling Assignment (where we may close some tasks), we commit all tasks.
3. When committing upon user-requested, we commit all tasks.

Pardon me if they are already covered, and just lmk :)

Reply:
Added new test for (1). Case (3) is covered by modified test `shouldMaybeCommitAllActiveTasksThatNeedCommit` (formally `shouldMaybeCommitActiveTasks`).

For case (2), I am not sure if we need to change anything? `handleAssignment()` is not changed in this PR, because we either resume a task or close it: if we resume a task there is nothing to be committed from my understanding and thus it seems sufficient to only commit tasks that we close?

Reply:
With 2) under eos-beta, when committing the txn we would avoidably commit the sent records from other tasks; so I'm wondering if we should also commit them as well, otherwise some outgoing records would be included in the txn while the incoming partition offsets would not be committed.

Reply:
Ack. My miss-conception was that a task would be in SUSPENDED state when we resume it (and thus it would have been committed already). However, it seems that `resume()` can also happen for a RUNNING task, for which case it's a no-op. Will fix it.

Reply:
Yeah with KIP-429, during the rebalance some tasks would still be processed and hence records being sent out; so the resuming tasks could be in RUNNING state as well.

Reply:
Ack. Fixed `TaskManager#handleAssignment()` and added new tests for case (2)